Emphasize pitfall with os-specific dependencies of noarch packages (#2860)
conda-forge/conda-forge.github.io: docs/maintainer/knowledge_base.mdx

2. `conda-forge.yml`'s [noarch_platforms](conda_forge_yml.mdx#noarch-platforms) option.
3. **conda-build 3.25.0 or above** changing the build hash depending on virtual packages used.

The idea is to generate **different noarch packages** for each OS needing different dependencies.
Let's say you have a pure Python package, perfectly eligible for `noarch: python`, but on Windows
it requires `windows-only-dependency`. You might have something like:

<RecipeTabs>

> ```recipe title="recipe/meta.yaml (non-noarch)"

Reviewer: ```suggestion
```recipe title="recipe/meta.yaml (arch-specific)"
```

Reviewer 2: Ah, sorry, I hadn't seen this. I don't care which one we use but it needs to stay consistent.

> ```recipe title="recipe/meta.yaml (non-noarch)"

Reviewer 2: ```suggestion
```recipe title="recipe/meta.yaml (arch-specific)"
```

> ```recipe title="recipe/meta.yaml (arch-specific)"

Reviewer 2: ```suggestion
```recipe title="recipe/meta.yaml (non-noarch)"
```

    - windows-only-dependency  # [win]
```

> ```recipe title="recipe/recipe.yaml (non-noarch)"

Reviewer 2: ```suggestion
```recipe title="recipe/recipe.yaml (arch-specific)"
```

Author: This is much better :star_struck:

</RecipeTabs>

:::warning[Important]
Do not forget to enumerate all platforms in your selectors, **including virtual package
dependencies**! For OS-specific builds, the solver requires virtual packages to be

> present in the package metadata in order to select the correct package for your

Reviewer 2: ```suggestion
present in the runtime requirements in order to select the correct package for your
```
